packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -518,6 +518,10 @@ private void handleAddView(View view) {
   private void handleRemoveView(View view) {
     UiThreadUtil.assertOnUiThread();
 
+    if (indexOfChild(view) == -1){
+      return;
+    }
+
     if (!customDrawOrderDisabled()) {
```

Comment:
Can you move this `indexOfChild` check inside this if-branch? That way we avoid the check in the new architecture.

Reply:
Thanks, that's a good idea